### ISSUE TITLE
Add: MimeType Correction and Whitelist

### ISSUE THREAD

**Issue by xErik** [expressjs/multer#148]
I receive files with the wrong mime type. That makes filtering by mime type difficult.

Ideas:

1. Add hook-method `mimetype` to determine correct mime type.
2. Add hook method `onMimeTypeLimit` to allow wrong mime type handling.

If my proposals below are interesting, I could give it a try and open a pull request. Although I am new to node.

# Determine correct mime type

Provide a hook-method called `mimetype` that gets called automatically by multer (before the file is written). This method `mimetype` is bound to return the correct mime type.

Currently, I call the method below manually to determine the mimetype. And rename the file manually afterwards.  

```javascript
var fileType = require('file-type');
var readChunk = require('read-chunk');

multer({
    mimetype: function(file) {
        // fileType() returns: { ext: 'mp4', mime: 'video/mp4' }
        var buffer = readChunk.sync(file.path, 0, 262);
        file.mimetype =  fileType(buffer).mime;
        file.extension =  fileType(buffer).ext;
    }
});
```

# Mime Type Whitelist / Blacklist

Provide a method `onMimeTypeLimit` which gets called automatically, if the mime type does not appear in `limits.onMimeTypeLimit` array.

```javascript
onMimeTypeLimit: function(file) {
    console.log('Deleting', '['+file.originalname+']', '['+file.mimetype+']');
    fs.unlink('./' + file.path); // delete the partially written file
},

limits: {
    onMimeTypeLimit: ['video/*']
}
```

**Comment by LinusU** [comment]
This would be fixed by #155, feel free to contribute to that discussion :)

Also, in you example you are trying to read the file before it has been accepted. That is the largest obstacle with this, I have addressed it in the other issue.

**Comment by xErik** [comment]
Yes, I let the upload complete, as I found no API function to access the first `n` bytes and test them for the mime type. It's a work around.
